hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -820,6 +963,129 @@ property m =
   Property defaultConfig $
     withFrozenCallStack (evalM m)
 
+------------------------------------------------------------------------
+-- Classification
+
+instance Semigroup Cover where
+  (<>) NoCover NoCover =
+    NoCover
+  (<>) _ _ =
+    Cover
+
+instance Monoid Cover where
+  mempty =
+    NoCover
+  mappend =
+    (<>)
+
+instance Semigroup CoverCount where
+  (<>) (CoverCount n0) (CoverCount n1) =
+    CoverCount (n0 + n1)
+
+instance Monoid CoverCount where
+  mempty =
+    CoverCount 0
+  mappend =
+    (<>)
+
+toCoverCount :: Cover -> CoverCount
+toCoverCount = \case
+  NoCover ->
+    CoverCount 0
+  Cover ->
+    CoverCount 1
+
+-- | This semigroup is right biased. The name, location and percentage from the
+--   rightmost `Classifier` will be kept. This shouldn't be a problem since the
+--   library doesn't allow setting multiple classes with the same 'ClassifierName'.
+instance Semigroup a => Semigroup (Classifier a) where
+  (<>) (Classifier _ _ _ m0) (Classifier name location percentage m1) =
+    Classifier name location percentage (m0 <> m1)
+
+instance Semigroup a => Semigroup (Coverage a) where
+  (<>) (Coverage c0) (Coverage c1) =
+    Coverage $
+      Map.foldrWithKey (Map.insertWith (<>)) c0 c1
+
+instance (Semigroup a, Monoid a) => Monoid (Coverage a) where
+  mempty =
+    Coverage mempty
+  mappend =
+    (<>)
+
+mkCoverage :: Maybe Span -> ClassifierName -> CoverPercentage -> Cover -> Coverage Cover
+mkCoverage mlocation name minimum_ cover_ =
+  Coverage $
+    Map.singleton name (Classifier name mlocation minimum_ cover_)
+
+coverPercentage :: TestCount -> CoverCount -> CoverPercentage
+coverPercentage (TestCount tests) (CoverCount count) =
```

Comment:
@felixmulder, couldn't we just write this as

```hs
coverPercentage :: TestCount -> CoverCount -> CoverPercentage
coverPercentage (TestCount tests) (CoverCount count) =
  CoverPercentage (fromIntegral count / fromIntegral tests * 100)
```

What `thousandths` means?

Reply:
I'm trying to troubleshoot [this](https://github.com/hedgehogqa/haskell-hedgehog/pull/288#discussion_r940676278), that was the reason I asked.